tests/window_management_tests/test_minimal_window_manager.cpp

```diff
@@ -694,6 +691,80 @@ TEST_F(MinimalWindowManagerWithThreeWindows,
     EXPECT_TRUE(is_above(windows[1], windows[0]));
 }
 
+TEST_F(MinimalWindowManagerWithOneWindow, minimized_window_returns_to_previous_state_on_focus)
+{
+    miral::WindowSpecification spec;
+    spec.state() = mir_window_state_minimized;
+    request_modify(windows[0], spec);
+    EXPECT_THAT(tools().info_for(windows[0]).state(), Eq(mir_window_state_minimized));
+
+    tools().select_active_window(windows[0]);
+    EXPECT_THAT(tools().info_for(windows[0]).state(), Eq(mir_window_state_restored));
+}
+
+TEST_F(MinimalWindowManagerWithOneWindow, restoring_a_minimized_window_with_no_active_window_focuses_the_restored_window)
+{
+    miral::WindowSpecification spec;
+    spec.state() = mir_window_state_minimized;
+    request_modify(windows[0], spec);
+    EXPECT_THAT(focused(windows[0]), Eq(false));
+
+    spec.state() = mir_window_state_restored;
+    request_modify(windows[0], spec);
+    EXPECT_THAT(focused(windows[0]), Eq(true));
+}
+
+TEST_F(MinimalWindowManagerWithThreeWindows, minimizing_an_inactive_window_maintains_its_place_in_the_z_order)
+{
+    miral::WindowSpecification spec;
+    spec.state() = mir_window_state_minimized;
+    request_modify(windows[1], spec);
+
+    EXPECT_THAT(is_above(windows[2], windows[1]), Eq(true));
+    EXPECT_THAT(is_above(windows[1], windows[0]), Eq(true));
+}
+
+TEST_F(MinimalWindowManagerWithThreeWindows,
+    minimizing_the_active_window_selects_a_new_window_and_moves_the_minimized_down_in_the_z_order)
+{
+    miral::WindowSpecification spec;
+    spec.state() = mir_window_state_minimized;
+    request_modify(windows[2], spec);
+
+    EXPECT_THAT(is_above(windows[1], windows[2]), Eq(true));
+    EXPECT_THAT(is_above(windows[2], windows[0]), Eq(true));
```

Comment:
This!

IMO minimized windows should go to the back (it is what happens on GNOME), so that when you Alt+Tab you get minimized windows at the end.

But not this PR :)